src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,63 @@
+package parser;
+
+import exception.DukeException;
+
+import java.util.ArrayList;
+
+public class Parser {
+    public static ArrayList<String> parse(String input) throws DukeException {
```

Comment:
This method is also quite long, around ~60 lines. Perhaps consider shortening it by refactoring some code to a different method.